internal/pkg/image/sif.go

```diff
@@ -31,7 +34,81 @@ func (f *sifFormat) initializer(img *Image, fileinfo os.FileInfo) error {
 	if bytes.Index(b, []byte(sifMagic)) == -1 {
 		return fmt.Errorf("SIF magic not found")
 	}
+
+	// Load the SIF file
+	fimg, err := sif.LoadContainerFp(img.File, !img.Writable)
+	if err != nil {
+		return err
+	}
+
+	// Get the default system partition image
+	part, _, err := fimg.GetPartPrimSys()
+	if err != nil {
+		return err
+	}
+
+	// record the fs type
+	fstype, err := part.GetFsType()
+	if err != nil {
+		return err
+	}
+	if fstype == sif.FsSquash {
+		img.Partitions[0].Type = SQUASHFS
+	} else if fstype == sif.FsExt3 {
+		img.Partitions[0].Type = EXT3
+	} else {
+		return fmt.Errorf("unknown file system type: %v", fstype)
+	}
+
+	img.Partitions[0].Offset = uint64(part.Fileoff)
+	img.Partitions[0].Size = uint64(part.Filelen)
+
+	// store all remaining sections
+	img.Sections = make([]Section, 0)
+
+	for _, desc := range fimg.DescrArr {
+		if ptype, err := desc.GetPartType(); err == nil {
+			// overlay partitions
+			if ptype == sif.PartOverlay && part.Groupid == desc.Groupid && desc.Used {
+				fstype, err := desc.GetFsType()
+				if err != nil {
+					continue
+				}
+				partition := Section{
+					Offset: uint64(desc.Fileoff),
+					Size:   uint64(desc.Filelen),
+					Name:   desc.GetName(),
+				}
+				switch fstype {
+				case sif.FsSquash:
+					partition.Type = SQUASHFS
+				case sif.FsExt3:
+					partition.Type = EXT3
+				}
+				img.Partitions = append(img.Partitions, partition)
+			}
+		} else {
+			// anything else
+			data := Section{
+				Offset: uint64(desc.Fileoff),
+				Size:   uint64(desc.Filelen),
+				Type:   uint32(desc.Datatype),
+				Name:   desc.GetName(),
+			}
+			img.Sections = append(img.Sections, data)
+		}
+	}
+
 	img.Type = SIF
+
+	// UnloadContainer close image, just want to unmap image
```

Comment:
Is this comment referencing why we're not using a `sif` package method to close the container? If so, no additional feedback here

Reply:
Must keep image file descriptor open, so just release mapped memory